typings/index.d.ts

```diff
@@ -7,7 +7,8 @@ export function renderHook<P, R>(
   } & RenderOptions
 ): {
   readonly result: {
-    current: R
+    readonly current: R,
+    readonly error: Error
```

Comment:
Again, technically you can throw anything, so maybe this should be a bit looser?

Also, typescript isn't really my thing, so not sure if `readonly` usage is correct here and the fact that either `current` or `error` will be undefined depending on whether the hook threw or not (I think types are nullable/undefinedable by default in Typescript?)?